### PR TITLE
Remove injected head scripts (close #2622)

### LINKED ISSUE #2622: Remove injected scripts from <head> after execution

### PR DESCRIPTION
**Pull request by Farfurix**
## Todo
- [x] [check these changes in TestCafe tests](https://github.com/DevExpress/testcafe/pull/6175)
- [x] check an `iframe` with the `sandbox` attribute
- [x] remove 'scripts' override
- [x] add a test for the `page-with-img` case
- [x] test name
- [x] fix "document.write for page html" client test

## Approach
Remove injected `script-hammerhead-shadow-ui` head scripts from DOM. NOTE: we don't need to change collection filtering, since its logic related to common shadow ui elements. Place removing logic at the beginning of the script (except for `initHammerheadClient`).

## Additional changes
Add a request hook test for the existing `page-with-img` case.

## Pre-Merge TODO
- [x] Write tests for your proposed changes
- [x] Make sure that existing tests do not fail

### CODE REVIEW

**Comment by Farfurix** [comment]
[testcafe-hammerhead-24.1.1.zip](https://github.com/DevExpress/testcafe-hammerhead/files/6374754/testcafe-hammerhead-24.1.1.zip)

File: src/client/index.js.wrapper.mustache [hunk 1]
```diff
@@ -1,3 +1,5 @@
 (function initHammerheadClient () {
     {{{source}}}
 })();
+
+window['%hammerhead%'].utils.removeInjectedHeadScript();
```
Comment (LavrovArtem) [inline src/client/index.js.wrapper.mustache:5]: I think that the `head` word is extra. Just use the `removeInjectedScript` function name.

**Review by LavrovArtem — CHANGES_REQUESTED** [review]

**Review by LavrovArtem — APPROVED** [review]

File: test/server/proxy/request-hooks-test.js [hunk 2]
```diff
@@ -719,6 +724,40 @@ describe('Request Hooks', () => {
             });
     });
 
+    it('TODO: naming', () => {
```
Comment (miherlosev) [inline test/server/proxy/request-hooks-test.js:727]:
```js
it('should proxy images if there are registered request filter rules', () {
   const url = 'http://127.0.0.1:2000/page-with-img'
   const rule = new RequestFilterRule(url);

  session.addRequestEventListeners(rule, {
       onRequest: noop
  });

   const options = {
            url:     proxy.openSession(url, session),
            headers: {
                accept: PAGE_ACCEPT_HEADER
            }
        };

   ...
});
```

**Review by miherlosev — CHANGES_REQUESTED** [review]

**Comment by Farfurix** [comment]
[testcafe-hammerhead-24.1.1.zip](https://github.com/DevExpress/testcafe-hammerhead/files/6403448/testcafe-hammerhead-24.1.1.zip)

**Comment by Farfurix** [comment]
[testcafe-hammerhead-24.2.0.zip](https://github.com/DevExpress/testcafe-hammerhead/files/6419213/testcafe-hammerhead-24.2.0.zip)

**Review by LavrovArtem — APPROVED** [review]

**Comment by Farfurix** [comment]
Functional tests are green: https://github.com/DevExpress/testcafe/pull/6175

**Review by miherlosev — APPROVED** [review]

**Pull request merged**